db/migrate/20220316205242_change_profile_rules_to_profile_links.rb

```diff
@@ -0,0 +1,17 @@
+class ChangeProfileRulesToProfileLinks < ActiveRecord::Migration[7.0]
+  class ProfileLink < ActiveRecord::Base
+  end
+
+  def change
+    rename_table :profile_rules, :profile_links
+    add_reference :profile_links, :entity, type: :uuid
+    add_column :profile_links, :type, :string
+    reversible do |migration|
+      migration.up do
+        ProfileLink.update_all 'entity_id=rule_id'
+        ProfileLink.update_all "type='ProfileRule'"
+      end
+    end
+    remove_column :profile_links, :rule_id, :uuid
```

Comment:
No need for the type, that's implicit:
```suggestion
    remove_column :profile_links, :rule_id
```